src/utils/dataStructureUtils.ts

```diff
@@ -1,12 +1,31 @@
 /**
  * Sort a map by key
- * @param map <string, { key: unknown }>
- * @Return sorted map
+ * @param map <string, unknown>
+ * @Return array of map's values sorted by key
  */
-function sortMapByKey(map: Map<string, { key: unknown }>) {
-  return (Array.from(map.values())).sort((obj1, obj2) => (obj1.key > obj2.key ? 1 : -1));
+function mapToSortedArray(map: Map<string, unknown>) {
+  const array = Array.from(map.entries()).sort();
+
+  return array.map((v) => v[1]);
+}
+
+/**
+ * Checks if `value` is classified as a `String` primitive or object.
+ * Based on lodash/isString
+ * @param {*} value The value to check.
+ * @returns {boolean} `true` if `value` is a string, else `false`.
+ */
+function isString(value: unknown) {
```

Comment:
I think this is an overkill to what we need, 

We could probably do something simpler 
`if (typeof myVar === 'string' || myVar instanceof String)`

We basically want to support `'foo'` and `new String('foo')` - I'm not even sure we should support `new String('foo')` to be honest.


(Taken from this stack overflow https://stackoverflow.com/questions/4059147/check-if-a-variable-is-a-string-in-javascript)

Reply:
ok